src/feedback/trulens/feedback/dummy/endpoint.py

```diff
@@ -273,128 +346,151 @@ def post(
                 # pretend to produce huggingface api classification results
-                return self._fake_classification()
+                ret = self._fake_classification()
             else:
-                return self._fake_completion(
-                    model=payload["model"],
-                    prompt=payload["prompt"],
-                    temperature=payload["temperature"],
+                ret = self._fake_completion(
+                    model=json["model"],
+                    prompt=json["prompt"],
+                    temperature=json["temperature"],
                 )
 
+            return self._fake_post_response(
+                status_code=200, json=ret, request=request
+            )
+
         else:
             raise RuntimeError("Unknown random result type.")
 
     def _fake_completion(
         self, model: str, prompt: str, temperature: float
-    ) -> Dict:
+    ) -> List[serial_utils.JSON]:
         generated_text: str = f"""
 First an integer: 2 . Also, this is my response to a prompt of length
 {len(prompt)} with a model {model} with temperature {temperature}. Also, here is
 an integer in case this is being used as a score: 2
 """
 
-        return {
-            "completion": generated_text,
-            "status": "success",
-            "usage": {
-                # Fake usage information.
-                "n_tokens": len(generated_text.split()) + len(prompt.split()),
-                "n_prompt_tokens": len(prompt.split()),
-                "n_completion_tokens": len(generated_text.split()),
-                "cost": len(generated_text) * 0.0002
-                + len(prompt.split()) * 0.0001,
-            },
-        }
+        return [
+            {
+                "completion": generated_text,
+                "status": "success",
+                "usage": {
+                    # Fake usage information.
+                    "n_tokens": len(generated_text.split())
+                    + len(prompt.split()),
+                    "n_prompt_tokens": len(prompt.split()),
+                    "n_completion_tokens": len(generated_text.split()),
+                    "cost": len(generated_text) * 0.0002
+                    + len(prompt.split()) * 0.0001,
+                },
+            }
+        ]
 
     def completion(
         self, *args, model: str, temperature: float = 0.0, prompt: str
-    ) -> Dict:
+    ) -> serial_utils.JSON:
         """Fake text completion request."""
 
+        # TODO: move this to provider
+
         # Fake http post request, might raise an exception or cause delays.
         return self.post(
             url="https://fakeservice.com/completion",
-            payload={
+            json={
                 "mode": "completion",
                 "model": model,
                 "prompt": prompt,
                 "temperature": temperature,
                 "args": args,  # include extra args to see them in post span
             },
-        )
+        ).json()[0]
```

Comment:
same concern as in other instances of using `json()[0]` - could we verify if indexing error is not going to happen here?

Reply:
`_fake_completion` and `_fake_classification` are the only status 200 returns in `DummyAPI` so here we can be sure of `[0]` existing. Added a check for status code just in case post returns without raising error on non-200 statuses.